opentelemetry-otlp/examples/basic-otlp-http/src/main.rs

```diff
@@ -71,51 +71,48 @@ fn init_metrics() -> Result<opentelemetry_sdk::metrics::SdkMeterProvider, Metric
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
-    let result = init_tracer_provider();
-    assert!(
-        result.is_ok(),
-        "Init tracer failed with error: {:?}",
-        result.err()
-    );
-
-    let tracer_provider = result.unwrap();
+    let tracer_provider = init_traces()?;
     global::set_tracer_provider(tracer_provider.clone());
```

Comment:
It might be better to set the global provider within the `init_traces` and `init_metrics` method itself. That could better convey to the users that setting global providers is required as part of the initialization setup.

Reply:
I wanted to show explicitly that it is cheap and okay to clone the provider here. 
Maybe setup another help create_provider, which is invoked from init_traces so we can cover your recommendation, and also show the cloning aspect..?